[Tejas Gandhi] iP

newway1814 opened the pull request:
# DuckBot - Task Management Chatbot
> "Management must manage!" -  Harold Geneen [(source)](https://www.inloox.com/company/blog/articles/the-62-most-inspiring-project-management-quotes/)

DuckBot is your solution for never missing another deadline.

Our 🦆🤖 is: 
+ ~~SIMPLE~~ *__EXTREMELY SIMPLE__* to navigate
+ User-Friendly
+ Lightning speed Performance.

__Impressive Features__:
- ✅ Manage Tasks Effectively
- 🕛Store the deadlines with accurate Date and Time
- 🔎Efficient Task Search
- User Interface Integration (Coming Soon)

Here is the Main method that starts our DuckBot: 
``` 
public static void main(String[] args) {
        new Duck("data/duck.txt").run();
    }

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

newway1814 committed:
Completed Level 0

newway1814 committed:
Completed Level-1

newway1814 committed:
Completed Level-2

newway1814 committed:
Completed Level-3

newway1814 committed:
Completed Level-4

newway1814 committed:
Completed Level-5

newway1814 committed:
Completed Level-6

newway1814 committed:
Completed A-TextUiTesting

newway1814 committed:
Completed Level-7

newway1814 committed:
Merge branch 'branch-Level-7'

newway1814 committed:
make illegal date format exception

newway1814 committed:
add basic date parsing functionality for deadline events

newway1814 committed:
fix merge conflicts

newway1814 committed:
fix datetime parse error in halting the program

newway1814 committed:
Applied OOP Principles

newway1814 committed:
Merge branch 'branch-A-OOP'

newway1814 committed:
Organised the code into packages

newway1814 committed:
Solve conflicts

newway1814 committed:
Merge branch 'add-gradle-support'

newway1814 committed:
Add gradle support

newway1814 committed:
Set up J-unit

newway1814 committed:
Set up jar file

newway1814 committed:
Add JavaDoc

newway1814 committed:
Merge branch 'A-JavaDoc'

newway1814 committed:
Complete Level-9

newway1814 committed:
Merge branch 'branch-level-9'

newway1814 committed:
Add Coding Standard

newway1814 committed:
Merge branch 'branch-A-CodingStandard'

newway1814 committed:
Add GuI

newway1814 committed:
Merge branch 'branch-Level-10'

jacobcuison commented on src/main/java/duck/IoHandler.java:
```diff
@@ -46,45 +45,51 @@ public String typeMessage() {
+        if (taskList.size() == 0) {
+            return "There are no tasks in your lists.";
+        }
+        result.append("Here are all the tasks in your list: \n");
```
I love how you used StringBuilder in this function, this is definitely something I can learn from you and will try to implement in my code :)

jacobcuison commented on src/main/java/duck/Main.java:
```diff
@@ -0,0 +1,27 @@
+import javafx.stage.Stage;
+
+public class Main extends Application {
+    private Duck duke = new Duck();
```
Nitpicking here, but perhaps you could rename the `duke` variable to `duck` :)

jacobcuison commented on src/main/java/duck/IoHandler.java:
```diff
@@ -46,45 +45,51 @@ public String typeMessage() {
-        System.out.println("Here are the matching tasks in your list: ");
+    public String displaySearchResults(ArrayList<Task> list) {
+        StringBuilder result = new StringBuilder();
+        if (list.isEmpty()) {
```
I like how you used this guard clause to prevent unnecessary nesting!

jacobcuison commented on src/main/java/duck/Duck.java:
```diff
@@ -1,39 +1,172 @@
+     */
+    @FXML
+    public String getResponse(String input) {
+        return parser.parse(input, ioHandler, taskList, storage).replace("\n", "\n\n");
```
Maybe you can add a comment here about the `.replace("\n", "\n\n")` part, I was a bit tripped up when I read it.

jacobcuison reviewed (commented):
Hi Tejas! Overall, your code was really easy to follow for me and I couldn't spot any major coding standard violations. I think yout code is of really great quality as well, and I learnt some cool techniques just by reading it. I just had some nitpicks but otherwise, looks good to me! All the best for this mod 👍

newway1814 committed:
Add Assertions for Code Reliabilty

In this commit, I have added a series of assert statements to increase
the reliability and robustness of the project.

Assertions help the programmers to catch bugs in early stages making it
easier for the programmers to fix the issue.

Key Changes:
-> Added assertions to TaskList and parser.
-> Checked for edge cases to ensure proper correctness of the logic.
-> Improved error handling by using assertion to prevent haywire
behaviour of the program.
More documentation on assertions please refer -> https://www.javatpoint.com/assertion-in-java

newway1814 committed:
Enhance Code Quality for improved directness

In this commit, I have focused on improving overall code quality.
Enhancing Code Quality is useful for better understanding of the code.

In this Commit I have focused on improving the overall code quality of
the project.

I have refactored complex functions into smaller, modular components for
better readability.
Appplied consistent coding conventions along with formatting the [pre
existing code to enhance code consistency.

newway1814 committed:
Merge pull request #1 from newway1814/branch-A-Assertions

Add Assertions for Code Reliabilty

newway1814 committed:
Merge pull request #2 from newway1814/branch-A-CodeQuality

Enhance Code Quality for improved directness

newway1814 committed:
Add Extension

newway1814 committed:
Merge branch 'Extension'

newway1814 committed:
Correct Styling

newway1814 committed:
Add Ui.png image

newway1814 committed:
Add User Guide

newway1814 committed:
Completed User Guide

newway1814 committed:
Merge branch 'branch-A-UserGuide'

newway1814 committed:
Update UserGuide format

newway1814 committed:
Update UserGuide (for html render)

newway1814 committed:
Update UserGuide (for html render part 2)

newway1814 committed:
Update UserGuide (for html render part 3)

newway1814 committed:
Create _config.yaml

newway1814 committed:
Update _config.yaml

newway1814 committed:
Delete docs/_config.yaml

newway1814 committed:
Create _config.yaml

newway1814 committed:
Update README.md

newway1814 committed:
Update README.md

newway1814 committed:
Add J-Unit Tests

newway1814 committed:
Fix Bugs

newway1814 committed:
Merge branch 'branch-A-Test'

newway1814 committed:
Update README.md

newway1814 committed:
Update README.md

newway1814 committed:
Fix Bugs

newway1814 committed:
Merge branch 'master' of https://github.com/newway1814/ip

newway1814 committed:
Fix Bugs

newway1814 committed:
Fix Bugs

newway1814 committed:
Merge branch 'branch-A-Cross'

newway1814 committed:
Fix Bugs

newway1814 committed:
Update README.md

newway1814 committed:
Fixed Bugs

newway1814 committed:
Fix Bugs

newway1814 committed:
fix bugs
